src/go/configgenerator/filtergen/grpc_metadata_scrubber.go

```diff
@@ -26,30 +27,28 @@ const (
 	GrpcMetadataScrubberFilterName = "com.google.espv2.filters.http.grpc_metadata_scrubber"
 )
 
-type GRPCMetadataScrubberGenerator struct {
-	// skipFilter indicates if this filter is disabled based on options and config.
-	skipFilter bool
-}
+type GRPCMetadataScrubberGenerator struct{}
 
-// NewGRPCMetadataScrubberGenerator creates the GRPCMetadataScrubberGenerator with cached config.
-func NewGRPCMetadataScrubberGenerator(serviceInfo *ci.ServiceInfo) *GRPCMetadataScrubberGenerator {
-	return &GRPCMetadataScrubberGenerator{
-		skipFilter: !serviceInfo.Options.EnableGrpcForHttp1,
+// NewGRPCMetadataScrubberFilterGensFromOPConfig creates a GRPCMetadataScrubberGenerator from
+// OP service config + descriptor + ESPv2 options. It is a FilterGeneratorOPFactory.
+func NewGRPCMetadataScrubberFilterGensFromOPConfig(serviceConfig *servicepb.Service, opts options.ConfigGeneratorOptions, params FactoryParams) ([]FilterGenerator, error) {
+	if !opts.EnableGrpcForHttp1 {
```

Comment:
Add logs here when the filter gen is not enabled.

Reply:
Done